### PR TITLE
Make (old) PowerAnalytics work with PSY4

### PR DESCRIPTION
**Pull request by GabrielKS**
Draft, the simulation that generates test data is still failing. I'm using the following patches to get this far:

- In `HydroPowerSimulations.jl/src/hydro_generation.jl` (this one should probably be committed):
```
-PSI.proportional_cost(cost::PSY.HydroGenerationCost , ::HydroEnergySurplusVariable, ::PSY.HydroGen, ::AbstractHydroReservoirFormulation)=PSY.get_energy_surplus_cost(cost)
-PSI.proportional_cost(cost::PSY.HydroGenerationCost , ::HydroEnergyShortageVariable, ::PSY.HydroGen, ::AbstractHydroReservoirFormulation)=PSY.get_energy_shortage_cost(cost)
+PSI.proportional_cost(cost::PSY.StorageCost , ::HydroEnergySurplusVariable, ::PSY.HydroGen, ::AbstractHydroReservoirFormulation)=PSY.get_energy_surplus_cost(cost)
+PSI.proportional_cost(cost::PSY.StorageCost , ::HydroEnergyShortageVariable, ::PSY.HydroGen, ::AbstractHydroReservoirFormulation)=PSY.get_energy_shortage_cost(cost)
```
- In `PowerSystems.jl/src/models/cost_functions/StorageCost.jl` (this one is just a kludge until something gets updated in PSI):
```
+get_variable(value::StorageCost) = get_discharge_variable_cost(value)
```

### CODE REVIEW

**Comment by GabrielKS** [comment]
The simulation and all old PA tests pass now, at the cost of removing simulation of hydro. The two patches above are still required.

File: src/get_data.jl [hunk 1]
```diff
@@ -441,30 +440,53 @@ function get_load_data(
...
+                key ->
+                    PSY.get_time_series_type(key) == PSY.Deterministic &&
+                        PSY.get_name(key) == "max_active_power",
+                PSY.get_time_series_keys(load),
+            )
+            @assert length(keys) == 1
+            my_resolution = first(keys).resolution
+            if isnothing(resolution)
+                resolution = my_resolution
+                len = (horizon isa Dates.Period) ? Int64(horizon / resolution) : horizon
```
Comment (jd-lara) [inline src/get_data.jl:462]: We have methods in PSY to handle this

**Review by jd-lara — CHANGES_REQUESTED** [review]
Update the time series handling using PSY4 correctly.

File: src/get_data.jl (hunk 1 above)
Comment (GabrielKS) [reply src/get_data.jl:462]: Is there a way to `get_time_series_values` passing in a `Period`?

**Comment by GabrielKS** [comment]
@daniel-thom is there a way to get a subset of a time series using a horizon `Period` as @jd-lara is requesting? I couldn't find one.

**Comment by daniel-thom** [comment]
> @daniel-thom is there a way to get a subset of a time series using a horizon `Period` as @jd-lara is requesting? I couldn't find one.

Nope, we missed this one. You can only pass horizon_count as `len`. It could be fixed relatively easily.

**Pull request merged**